modules/certificate/certificate.service.js

```diff
@@ -42,38 +49,69 @@ class CertificatesService {
     return certificate;
   }
 
-  async addCertificate(name, value, id) {
-    const certificateExists = await Certificate.findOne({ name });
+  async generateCertificate(certificateData, userId, userRole) {
+    const randomNum = randomInt(100000, 999999);
+    const name = `hor${randomNum}`;
 
-    if (certificateExists) {
-      throw new RuleError(CERTIFICATE_ALREADY_EXISTS, BAD_REQUEST);
+    certificateData.name = name;
+
+    if (userRole === USER) {
+      certificateData.ownedBy = userId;
+    } else if (userRole === ADMIN || userRole === SUPERADMIN) {
```

Comment:
just else

Reply:
ok, but wouldn't we lose the readability of the code?